src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private final ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private final TruckProducer truckProducer = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machineList = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            machineList.addAll(bulldozerProducer.get());
+        }
+        if (type.equals(Excavator.class)) {
+            machineList.addAll(excavatorProducer.get());
+        }
+        if (type.equals(Truck.class)) {
+            machineList.addAll(truckProducer.get());
+        }
+        return machineList;
```

Comment:
The `getAll` method can be simplified by using a local variable for the producer. Instead of calling `addAll` on the `machineList` for each type, you could determine the appropriate producer first and then add all its machines to the list. This way, the code would be cleaner and easier to maintain.